GCRCatalogs/dc2_coadd.py

```diff
@@ -34,91 +84,303 @@ def _subclass_init(self, **kwargs):
 
         if not os.path.isdir(self._base_dir):
             raise ValueError('`base_dir` {} is not a valid directory'.format(self._base_dir))
-        self._datasets, self._columns = self._generate_native_datasets_and_columns()
-        if not self._datasets:
-            raise RuntimeError('No catalogs were found in `base_dir` {}'.format(self._base_dir))
 
         self._dataset_cache = dict()
 
-        self._quantity_modifiers = {
+        self._quantity_modifiers = self._generate_modifiers()
+        self._datasets, self._columns = self._generate_datasets_and_columns()
+        if not self._datasets:
+            err_msg = 'No catalogs were found in `base_dir` {}'
+            raise RuntimeError(err_msg.format(self._base_dir))
+
+    @staticmethod
+    def _generate_modifiers():
+        """Creates a dictionary relating native and homogenized column names
+
+        Returns:
+            A dictionary of the form {<homogenized name>: <native name>, ...}
+        """
+        modifiers = {
+            'objectId': 'id',
+            'parentObjectId': 'parent',
             'ra': (np.rad2deg, 'coord_ra'),
             'dec': (np.rad2deg, 'coord_dec'),
+            'centroidX': 'slot_Centroid_x',
+            'centroidY': 'slot_Centroid_y',
+            'centroidX_err': 'slot_Centroid_xSigma',
+            'centroidY_err': 'slot_Centroid_ySigma',
+            'centroid_flag': 'slot_Centroid_flag',
+            'psNdata': 'base_PsfFlux_area',
+            'extendedness': 'base_ClassificationExtendedness_value',
+            'HSM_res': 'ext_shapeHSM_HsmShapeRegauss_resolution',
+            'HSM_ell': (np.hypot, 'ext_shapeHSM_HsmShapeRegauss_e1', 'ext_shapeHSM_HsmShapeRegauss_e2'),
         }
 
+        modifiers['pass_basic_flag_cut'] = (
+            create_basic_flag_mash,
+            'deblend_skipped',
+            'base_PixelFlags_flag_edge',
+            'base_PixelFlags_flag_interpolatedCenter',
+            'base_PixelFlags_flag_saturatedCenter',
+            'base_PixelFlags_flag_crCenter',
+            'base_PixelFlags_flag_bad',
+            'base_PixelFlags_flag_suspectCenter',
+            'base_PixelFlags_flag_clipped',
+            'ext_shapeHSM_HsmShapeRegauss_flag',
```

Comment:
I think our focus is on DC2 for now.  I hear your point about maintain something general, and perhaps there's some inheritance one could set up, but there will also be these small differences in different runs that we want to encode in our reader and not make each user have to know.

Reply:
But I realize now, that perhaps we should remove that flag because it may not do the right thing for stars.

Reply:
it may fail for some stars, not sure... but the point is that this flag is mostly useful if you are going t use the HSM shapes, like for a lensing sample, it may introduce unnecessary/unwanted selections for other use cases